enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -62,17 +62,22 @@
     (eduction (map #(serdes/extract-all % extract-opts)) cat (model-set opts))))
 
 ;; TODO Properly support "continue" - it should be contagious. Eg. a Dashboard with an illegal Card gets excluded too.
-(defn- descendants-closure [targets]
+(defn- chase-for [targets func]
   (loop [to-chase (set targets)
          chased   #{}]
     (let [[m i :as item] (first to-chase)
-          desc           (serdes/descendants m i)
+          res            (func m i)
           chased         (conj chased item)
-          to-chase       (set/union (disj to-chase item) (set/difference desc chased))]
+          to-chase       (set/union (disj to-chase item) (set/difference res chased))]
       (if (empty? to-chase)
         chased
         (recur to-chase chased)))))
 
+(defn- gather-lineage [targets]
+  (set/union
+   (chase-for targets serdes/ascendants)
+   (chase-for targets serdes/descendants)))
```

Comment:
`lineage` is more misleading than helpful, and I can't think of a better name. I think you could just inline this in escape analysis